src/MCPClient/lib/clientScripts/has_packages.py

```diff
@@ -24,14 +24,29 @@ def is_extractable(f):
         return False
 
 
+AM_DATE_DELIMITER = '-AM-DATE-SFX-'
```

Comment:
Reading your rationale I like this approach. As to where values like this should sit, do you think there is potential to have them at a more central location across the codebase (archivematicaCommon?) or if there is a place where Archivematica 'special values'/keywords could exist so that they can be looked up? E.g. in validation operations to make sure user's don't accidentally transfer values with keywords in them - or that we know they're there and can work around them as appropriate? 

I don't want to expand the scope of this fix too much though, and I am still getting to grips with the impact of what might happen say if we wanted to actually transfer a file with a `processingMCP.xml` or `metadata` folder in them, but have those treated as objects not special files/dirs in the system.

Reply:
I see what you mean, it would be good for future work to define these in a central location. However, I don't see anything in archivematicaCommon/lib/ or MCPClient/lib/clientScripts/ where a precedent for something like this has been set. I guess we could introduce a constants.py file somewhere, but I'm inclined to leave this as is for now.

Reply:
I see what's bothering you though. The interconnectedness of `already_extracted` and `suffix_date` is something that should be enforced or at least more explicitly expressed than in shared use of a constant. I'm wondering if this might be a case where property-based testing (Hypothesis: https://hypothesis.readthedocs.io/en/latest/) might be useful...